modules/gapi/src/executor/gexecutor.cpp

```diff
@@ -155,14 +155,23 @@ void cv::gimpl::GExecutor::run(cv::gimpl::GRuntimeArgs &&args)
 #if !defined(GAPI_STANDALONE)
             // Building as part of OpenCV - follow OpenCV behavior
             // if output buffer is not enough to hold the result, reallocate it
-            auto& out_mat   = *get<cv::Mat*>(args.outObjs.at(index));
-            createMat(desc, out_mat);
+            if (cv::util::holds_alternative<cv::Mat*>(args.outObjs.at(index)))
+            {
+                auto& out_mat = *get<cv::Mat*>(args.outObjs.at(index));
+                createMat(desc, out_mat);
+            }
+            else
+            {
+                auto& out_mat = *get<cv::gapi::own::Mat*>(args.outObjs.at(index));
```

Comment:
I'd still place these common lines for own::Mat in a lambda. Also this behavior of Ocv mode doesn't correlate with the comment above (on reallocation), please add a comment on own::Mat related behavior (never reallocated, checked to perfectly fit required meta)

Reply:
Fixed